From beb0d74195ddc642524fe1f5a2d65d944527d1b5 Mon Sep 17 00:00:00 2001
From: Alessandro Portale <alessandro.portale@qt.io>
Date: Wed, 13 Nov 2019 00:02:50 +0100
Subject: [PATCH] QSvgRenderer: Allow to remove a viewBox

m_implicitViewBox needs to be true if the explicit viewBox is null. That
relationship between m_viewBox and m_implicitViewBox can be see in
QSvgTinyDocument::viewBox().

Task-number: QTBUG-79933
Change-Id: I6ea5a849479b10117128bcaf1799e3770b145e77
Reviewed-by: Simon Hausmann <simon.hausmann@qt.io>
---
 src/svg/qsvgtinydocument.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/svg/qsvgtinydocument.cpp b/src/svg/qsvgtinydocument.cpp
index 7e355086..a2bf1c77 100644
--- a/src/svg/qsvgtinydocument.cpp
+++ b/src/svg/qsvgtinydocument.cpp
@@ -338,7 +338,7 @@ void QSvgTinyDocument::setHeight(int len, bool percent)
 void QSvgTinyDocument::setViewBox(const QRectF &rect)
 {
     m_viewBox = rect;
-    m_implicitViewBox = false;
+    m_implicitViewBox = rect.isNull();
 }
 
 void QSvgTinyDocument::addSvgFont(QSvgFont *font)
-- 
2.16.3

